pylint/checkers/base/basic_checker.py

```diff
@@ -444,6 +451,14 @@ def visit_expr(self, node: nodes.Expr) -> None:
             self.add_message("pointless-string-statement", node=node)
             return
 
+        # Warn W0133 for exceptions that are used as statements
+        if isinstance(expr, nodes.Call):
+            inferred = utils.safe_infer(expr)
```

Comment:
Inference can be expensive, especially if the lru cache has reached is maxsize and old entries are overwritten.

We should not call for `safe_infer` for every `Call` node. Instead I would recommend to use a heuristic and filter likely candidates first.
* Almost all builtin error names contain either `Exception`, `Error`, or `Warning`. That likely applies to custom exceptions as well.
* This could  be further improved with a check against other predefined names. For the stdlib, those would probably be `KeyboardInterrupt`, `StopAsyncIteration`, `StopIteration`.
* Lastly a new config option could be added to allow the user to specify additional names (maybe even regex).

The goal IMO should be to capture 95% of the likely errors without the huge performance impact.

Reply:
Outch thank you for the check @cdce8p this seems like a show stopper for 2.16.0 release and a great advertisement for better benchmarking in CI.

Reply:
Thank you, @cdce8p!

I can take a look at this within the next 24h or so.  The name-based heuristics sound straightforward to implement - I have less confidence about the config option though, since I don't know much about how those work in `pylint` yet.

Is it fairly straightforward to re-run the benchmarking when a changeset is ready?

None of this should prevent either reverting the change, or someone else picking up the performance improvements if I don't get around to them soon.

Reply:
> I can take a look at this within the next 24h or so. The name-based heuristics sound straightforward to implement - I have less confidence about the config option though, since I don't know much about how those work in `pylint` yet.

I would suggest to start with the name check first and implement that. It should get you most of the way.
Adding a new config option isn't too difficult though. As example, you can take a look at this
https://github.com/PyCQA/pylint/blob/b87cf1631f9fcef278895382481cc0169cb2301e/pylint/checkers/design_analysis.py#L389-L398
https://github.com/PyCQA/pylint/blob/b87cf1631f9fcef278895382481cc0169cb2301e/pylint/checkers/design_analysis.py#L413-L415

Feel free to ask if you have questions.

> Is it fairly straightforward to re-run the benchmarking when a changeset is ready?

I've set up a semi-automatic pipeline for that in my fork. Probably easiest if you ping me for it.
If you really want to test it on your own, you could fork my home-assistant fork and take a look at https://github.com/cdce8p/ha-core/pull/11. Although I'm fairly confident that any changes here should be enough to remedy the issue and it's probably much easier for me to run the test again.